### PR TITLE
fix(std): code flow

### PR DESCRIPTION
**Pull request by behnammodi**
We don't need to run some code before dest exist condition

### CODE REVIEW

**Comment by behnammodi** [comment]
@bartlomieju Can you help me? I don't understand what's wrong

File: std/fs/ensure_symlink.ts [hunk 1]
```diff
@@ -30,6 +27,9 @@ export async function ensureSymlink(src: string, dest: string): Promise<void> {
 
   await ensureDir(path.dirname(dest));
 
+  const srcStatInfo: string = await Deno.lstat(src);
+  const srcFilePathType: string = getFileInfoType(srcStatInfo);
```
Comment (bartlomieju) [inline std/fs/ensure_symlink.ts:31]:
```
error: TS2322 [ERROR]: Type 'FileInfo' is not assignable to type 'string'.
  const srcStatInfo: string = await Deno.lstat(src);
        ~~~~~~~~~~~
    at file:///home/runner/work/deno/deno/std/fs/ensure_symlink.ts:30:9

TS2322 [ERROR]: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.
  const srcFilePathType: string = getFileInfoType(srcStatInfo);
        ~~~~~~~~~~~~~~~
    at file:///home/runner/work/deno/deno/std/fs/ensure_symlink.ts:31:9

TS2345 [ERROR]: Argument of type 'string' is not assignable to parameter of type 'FileInfo'.
  const srcFilePathType: string = getFileInfoType(srcStatInfo);
                                                  ~~~~~~~~~~~
    at file:///home/runner/work/deno/deno/std/fs/ensure_symlink.ts:31:51

TS2322 [ERROR]: Type 'FileInfo' is not assignable to type 'string'.
  const srcStatInfo: string = Deno.lstatSync(src);
        ~~~~~~~~~~~
    at file:///home/runner/work/deno/deno/std/fs/ensure_symlink.ts:63:9

TS2322 [ERROR]: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.
  const srcFilePathType: string = getFileInfoType(srcStatInfo);
        ~~~~~~~~~~~~~~~
    at file:///home/runner/work/deno/deno/std/fs/ensure_symlink.ts:64:9

TS2345 [ERROR]: Argument of type 'string' is not assignable to parameter of type 'FileInfo'.
  const srcFilePathType: string = getFileInfoType(srcStatInfo);
                                                  ~~~~~~~~~~~
    at file:///home/runner/work/deno/deno/std/fs/ensure_symlink.ts:64:51

Found 6 errors.
```

These are wrong types

Comment (behnammodi) [reply std/fs/ensure_symlink.ts:31]: Thanks a lot

File: std/fs/ensure_symlink.ts [hunk 2]
```diff
@@ -30,6 +27,11 @@ export async function ensureSymlink(src: string, dest: string): Promise<void> {
 
   await ensureDir(path.dirname(dest));
 
+  const srcStatInfo: Deno.FileInfo | undefined = await Deno.lstat(src);
+  const srcFilePathType: PathType | undefined = getFileInfoType(
+    srcStatInfo,
+  );
```
Comment (bartlomieju) [inline std/fs/ensure_symlink.ts:33]: @behnammodi what's the reason for adding explicit types here? It's more code and I'm sure IDEs infer it without a problem. Please revert.

File: std/fs/ensure_symlink.ts [hunk 3]
```diff
@@ -63,6 +62,11 @@ export function ensureSymlinkSync(src: string, dest: string): void {
 
   ensureDirSync(path.dirname(dest));
 
+  const srcStatInfo: Deno.FileInfo | undefined = Deno.lstatSync(src);
+  const srcFilePathType: PathType | undefined = getFileInfoType(
+    srcStatInfo,
+  );
```
Comment (bartlomieju) [inline std/fs/ensure_symlink.ts:68]: Ditto

**Review by bartlomieju — CHANGES_REQUESTED** [review]

**Comment by bartlomieju** [comment]
@behnammodi thanks for the PR, but I'm gonna close it without a merge. It's just moving code around and there are no tests that check any change in behavior.

**Comment by behnammodi** [comment]
@bartlomieju Ok, go-ahead 👍🏻

**Pull request closed without merging**